src/codegen/mod.rs

```diff
@@ -265,23 +277,31 @@ impl CodeGenerator for Var {
                ctx: &BindgenContext,
                result: &mut CodegenResult,
                item: &Item) {
-        let name = item.canonical_name(ctx);
+        let canonical_name = item.canonical_name(ctx);
+
+        // TODO: Maybe warn here if there's a type mismatch
```

Comment:
Thank you for your comment.
Do you mean the TODO is not necessary?
I guess it would result in a compile error in the C or C++ code.